libclamav_rust/src/onenote.rs

```diff
@@ -0,0 +1,355 @@
+
+        Some(file_data)
+    }
+
+    pub fn next_file(&mut self) -> Option<ExtractedFile> {
+        debug!("Looking to extract file from OneNote section...");
+
+        let mut file_data: Option<Vec<u8>> = None;
+
+        let remaining = if let Some(remaining_in) = self.remaining {
+            let file_data_store_object: &[u8] = &[
+                0xE7, 0x16, 0xE3, 0xBD, 0x65, 0x26, 0x11, 0x45, 0xA4, 0xC4, 0x8D, 0x4D, 0x0B, 0x7A,
+                0x9E, 0xAC,
+            ];
+
+            let remaining = if let Some(pos) = find_bytes(remaining_in, file_data_store_object) {
+                let (_, remaining) = remaining_in.split_at(pos);
+                // Found file data store object.
+                remaining
+            } else {
+                return None;
+            };
+
+            let data_length_bytes = &remaining[16..20];
+            let data_length = ((data_length_bytes[3] as u64) << 24)
+                | ((data_length_bytes[2] as u64) << 16)
+                | ((data_length_bytes[1] as u64) << 8)
+                | (data_length_bytes[0] as u64);
+
+            let data: &[u8] = &remaining[mem::size_of::<FileDataHeader>()
+                ..mem::size_of::<FileDataHeader>() + data_length as usize];
+
+            file_data = Some(data.to_vec());
+
+            Some(
+                &remaining
+                    [mem::size_of::<FileDataHeader>() + (data_length as usize)..remaining.len()],
+            )
+        } else {
+            None
+        };
+
+        self.remaining = remaining;
+
+        if let Some(file_data) = file_data {
+            Some(ExtractedFile {
+                name: None,
+                data: file_data,
+            })
+        } else {
+            None
+        }
+    }
+
+    pub fn next_file_vec(&mut self) -> Option<ExtractedFile> {
+        debug!("Looking to extract file from OneNote section...");
+
+        let mut file_data: Option<Vec<u8>> = None;
+
+        self.remaining_vec = if let Some(ref remaining_vec) = self.remaining_vec {
+            let file_data_store_object: &[u8] = &[
+                0xE7, 0x16, 0xE3, 0xBD, 0x65, 0x26, 0x11, 0x45, 0xA4, 0xC4, 0x8D, 0x4D, 0x0B, 0x7A,
+                0x9E, 0xAC,
+            ];
+
+            let remaining = if let Some(pos) = find_bytes(remaining_vec, file_data_store_object) {
+                let (_, remaining) = remaining_vec.split_at(pos);
+                // Found file data store object.
+                remaining
+            } else {
+                return None;
+            };
+
+            let data_length_bytes = &remaining[16..20];
+            let data_length = ((data_length_bytes[3] as u64) << 24)
+                | ((data_length_bytes[2] as u64) << 16)
+                | ((data_length_bytes[1] as u64) << 8)
+                | (data_length_bytes[0] as u64);
+
+            let data: &[u8] = &remaining[mem::size_of::<FileDataHeader>()
```

Comment:
If it makes the code more readable, you could define:
``` rust
const SIZE_OF_FILE_DATA_HEADER: usize = mem::size_of::<FileDataHeader>;
```
near the top of the file.  `mem::size_of` is a const fn, so you can do that.  It doesn't need to be runtime.

Reply:
Yeah tis a bit better. Thanks